packages/addons-inject/index.js

```diff
@@ -0,0 +1,309 @@
+  }
+
+  try {
+    const transformed = await transformResponse(response);
+    // Wait on the transformed body for errors to evaluate. Timeout errors and
+    // errors thrown during transform aren't actually raised until the body is
+    // read. If there was no return, we should return the original response.
+    if (transformed !== undefined) {
+      return new Response(await transformed.arrayBuffer(), transformed);
+    }
+  } catch (error) {
+    console.error("Discarding error:", error);
+    console.debug("Returning original response to avoid blank page");
+  }
+
+  return response;
+}
+
+export default {
+  fetch: onFetch,
+};
+
+/** Transform HTTP reponses
+ *
+ * This supports several versions of element removal and transformation of some
+ * content that was previously injected into project builds.
+ *
+ * Important: wait until the last minute to use a `response.clone()`, as if you
+ * create a clone and do not use the body, this consumes extra memory in the
+ * worker.
+ *
+ * @param response {Response} - The origin response
+ * @returns {Response} - If transformed, return a response, otherwise `null`.
+ **/
+async function transformResponse(response) {
+  const { headers } = response;
+
+  // Get the content type of the response to manipulate the content only if it's HTML
+  const contentType = headers.get("content-type") || "";
+  const injectHostingIntegrations =
+    headers.get("x-rtd-hosting-integrations") || "false";
+  const forceAddons = headers.get("x-rtd-force-addons") || "false";
+  const httpStatus = response.status;
+
+  // Log some debugging data
+  console.log(`ContentType: ${contentType}`);
+  console.log(`X-RTD-Force-Addons: ${forceAddons}`);
+  console.log(`X-RTD-Hosting-Integrations: ${injectHostingIntegrations}`);
+  console.log(`HTTP status: ${httpStatus}`);
+
+  // Debug mode for some test cases. This is just for triggering an exception
+  // from tests. There might be a way to conditionally enable this, but I had
+  // trouble getting Wrangler vars to work at all.
+  const throwError = headers.get("X-RTD-Throw-Error");
+  if (throwError) {
+    console.log(`Throw error: ${throwError}`);
+  }
+
+  // Get project/version slug from headers inject by El Proxito
+  const projectSlug = headers.get("X-RTD-Project") || "";
+  const versionSlug = headers.get("X-RTD-Version") || "";
+  const resolverFilename = headers.get("X-RTD-Resolver-Filename") || "";
+
+  // Check to decide whether or not inject Addons library. We only do this for
+  // `text/html` content types.
+  if (contentType.includes("text/html")) {
+    // Remove old implementation of our flyout and inject the new addons if the following conditions are met:
+    //
+    // - header `X-RTD-Force-Addons` is present (user opted-in into new beta addons)
+    // - header `X-RTD-Hosting-Integrations` is not present (added automatically when using `build.commands`)
+    //
+    if (forceAddons === "true" && injectHostingIntegrations === "false") {
```

Comment:
From https://github.com/readthedocs/readthedocs.org/blob/3b947be7e3603f4fbac15b6af9d1d441fa41db1d/readthedocs/proxito/middleware.py#L273-L339, looks like these headers are enabled one at the time (they both can't be true).

```suggestion
    // Remove old implementation of our flyout and inject the new addons if the following conditions are met:
    //
    // - header `X-RTD-Force-Addons` is present (user opted-in into new beta addons)
    // - header `X-RTD-Hosting-Integrations` is not present (added automatically when using `build.commands`)
    //
    if (forceAddons === "true") {
```

docstring should be updated as well.

Reply:
You're probably right, but I don't want to change these conditionals yet. I'm a little confused by the usage of both these headers.

Reply:
Yup agreed.

I do think we should consolidate or at least rename these. It's not clear what the difference is between `forceAddons` and `injectHostingIntegrations`, the naming is confusing. I'm guessing these, and respective headers, should have been closer to `areAddonsEnabled` and `isUsingBuildCommands`.